### PR TITLE
Test framework for evaluating if a backup integration works properly

### PR DESCRIPTION
**Pull request by ShellyKa13**
**What this PR does / why we need it**:
The PR improved tests framework by handling namespace creation and deletion in the general test framework instead in each test context. Handle clients creation better.
Create yaml that can be used also for other backup/restore providers to test if their tools integrates properly with kubevirt.
Adjust test to consume this yaml to make sure the yamls will not go old an outdated without notice.
Add some more test for backup and restore.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Add yamls to be used to make sure backup integration works properly
```

### CODE REVIEW

File: hack/build/run-functional-tests.sh [hunk 1]
```diff
@@ -58,4 +81,4 @@ velero_path=$(pwd)/${VELERO_DIR}
 (
     cd ${TESTS_DIR}
     KUBECONFIG=${kubeconfig_arg} PATH=${PATH}:${velero_path} ${test_command}
-)
```
Comment (akalenyu) [inline hack/build/run-functional-tests.sh:61]: no newline?

File: tests/framework/framework.go [hunk 2]
```diff
@@ -96,28 +137,107 @@ func getMaxFailsFromEnv() int {
...
+			}
+			ginkgo.By(fmt.Sprintf("Destroying namespace %q for this suite.", ns.Name))
+			err := DeleteNS(f.K8sClient, ns.Name)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}
+	}()
+
+	if ginkgo.CurrentGinkgoTestDescription().Failed {
+		f.reporter.FailureCount++
+		f.reporter.Dump(f.K8sClient, f.KvClient, ginkgo.CurrentGinkgoTestDescription().Duration)
```
Comment (akalenyu) [inline tests/framework/framework.go:188]: might want to consider https://github.com/kubevirt/containerized-data-importer/pull/2485
for easier debugging

**Review by akalenyu — COMMENTED** [review]
looks good
watch out for returning `false, err` in polling since it will just back out on random errors
we get bitten by this over in CDI

File: tests/framework/framework.go (hunk 2 above)
Comment (ShellyKa13) [reply tests/framework/framework.go:188]: nice

File: hack/build/run-functional-tests.sh (hunk 1 above)
Comment (ShellyKa13) [reply hack/build/run-functional-tests.sh:61]: its a redundant new line at the end of the file..

File: tests/framework/storage.go [hunk 3]
```diff
@@ -156,6 +156,19 @@ func DeleteDataVolume(kvClient kubecli.KubevirtClient, namespace, name string) e
...
+func DeleteDataVolumeWithoutDeletingPVC(kvClient kubecli.KubevirtClient, namespace, name string) error {
+	propagationOrphan := metav1.DeletePropagationOrphan
+	return wait.PollImmediate(pollInterval, waitTime, func() (bool, error) {
+		err := kvClient.CdiClient().CdiV1beta1().DataVolumes(namespace).Delete(context.TODO(), name, metav1.DeleteOptions{
+			PropagationPolicy: &propagationOrphan,
+		})
+		if err == nil || apierrs.IsNotFound(err) {
+			return true, nil
+		}
+		return false, err
```
[the remark this answers is not in the document]
Comment (ShellyKa13) [reply tests/framework/storage.go:168]: thanks

File: hack/build/run-functional-tests.sh (hunk 1 above)
Comment (akalenyu) [reply hack/build/run-functional-tests.sh:61]: Empty newline is the standard across our repos AFAIK

**Review by mhenriks — COMMENTED** [review]
This looks really awesome! 

Wondering about tests for VMI and VMIRS?

Also, what about DV garbage collection on vs off?

**Comment by ShellyKa13** [comment]
@mhenriks Thanks!
I checked and there are several standalone VMI tests which I can convert to also consume yamls and create appropriate yamls.
Regarding VMIRS I understand its missing need to also add relevant backup/restore action items so it will be done in different PR.
Same for GC, I working on adjusting all the tests to work both with and without GC. will be on different PR.

**Comment by ShellyKa13** [comment]
fixed @akalenyu comments.
Converted the positive standalone VMI test to consume yaml like @mhenriks asked.
GC and VMIRS will be in different PRs.

**Comment by mhenriks** [comment]
Great work @ShellyKa13!

/lgtm
/approve

**Pull request merged**